lib/rails_multisite/connection_management.rb

```diff
@@ -121,14 +125,28 @@ def self.host(env)
       end
     end
 
-    attr_reader :config_filename
+    def self.handler_key(spec)
+      if ActiveRecord::Base.respond_to?(:writing_role)
```

Comment:
This is a new API exposed in Rails 6.

Reply:
maybe just memoize the suffix, no need to call these methods over and over 

def self.handler_suffix
   @handler_suffix ||= ....
end